src/lightning/data/cache/writer.py

```diff
@@ -0,0 +1,273 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+from typing import Any, Dict, Optional
+
+import numpy as np
+
+from lightning.data.cache.compression import _COMPRESSORS
+from lightning.data.cache.pytree import tree_flatten, treespec_dumps
+from lightning.data.cache.serializers import _SERIALIZERS, Serializer
+from lightning.data.cache.worker import get_worker_info
+from lightning.data.datasets.env import _DistributedEnv, _WorkerEnv
+
+
+def cloud_path(cache_dir: str) -> Optional[str]:
+    cluster_id = os.getenv("LIGHTNING_CLUSTER_ID", None)
+    project_id = os.getenv("LIGHTNING_CLOUD_PROJECT_ID", None)
+    cloud_space_id = os.getenv("LIGHTNING_CLOUD_SPACE_ID", None)
+
+    if cluster_id is None or project_id is None or cloud_space_id is None:
+        return None
+    return f"s3://{cluster_id}/projects/{project_id}/cloudspaces/{cloud_space_id}/content/{cache_dir}/"
```

Comment:
we probably don't want our internal bucket structure exposed. Potential security risk and also gives away a lot of things. So far we have explicitly chosen not to do this in the `lightning.data` package pre GA

Reply:
I don't think it is a particular security risks as you still need access but yeah, we would need to access it in another way.